### ISSUE TITLE
about benchmark issue

### ISSUE THREAD

**Issue by konioyxgq** [pytorch/pytorch#45053]
Under the condition of benchmark=true, the training is repeated twice, and each training epoch=50, the model trained twice has a much worse effect on the test set. I know that in the case of benchmark=true, it is not reproducible during training, but the result is a bit worse, and the recall rate has dropped from 40.3% to 39.5%. Is this within the expected range?

**Comment by malfet** [comment]
Can you please give a bit more details, namely exact command you are trying to run, what vesion of PyTorch are you using and so on.

**Comment by konioyxgq** [comment]
In the script: 'torch.backends.cudnn.benchmark = True', 'torch.backends.cudnn.deterministic = True'
pytorch1.5+cu92

Distributed training is used:
```
import torch.distributed as dist
def main():
    cfg = configurations[1]
    ngpus_per_node = torch.cuda.device_count()
    world_size = cfg['WORLD_SIZE']
    cfg['WORLD_SIZE'] = ngpus_per_node * world_size
    mp.spawn(main_worker, nprocs=ngpus_per_node, args=(ngpus_per_node, cfg))
    
def main_worker(gpu, ngpus_per_node, cfg):

    SEED = cfg['SEED'] # random seed for reproduce results
    torch.manual_seed(SEED)
    torch.cuda.manual_seed_all(SEED)
    torch.backends.cudnn.benchmark = True
    torch.backends.cudnn.deterministic = True
```

**Comment by konioyxgq** [comment]
I know that when cudnn.benchmark=true, different convolution implementation algorithms are selected for calculation, so when cudnn.benchmark=true, the training process cannot be reproduced. But I think the results obtained by different convolution implementation algorithms are different but should not be very large. I found in the test that the recall rate of the test set obtained by repeated training twice dropped from 40.2% to 39.5%, and the error exceeded expectations. So I want to know if this error may be caused by other reasons when the benchmark=true, which can be avoided. In addition, is this error within the expected range?

**Comment by ptrblck** [comment]
> But I think the results obtained by different convolution implementation algorithms are different but should not be very large.

That's a valid assumption. 

Could you update to the latest stable PyTorch version (`1.6`) with CUDA10.2, which would ship with cudnn7.6.5.32 and rerun the script, please? (your cudnn version should be 7.6.3.30, if I'm not mistaken)

Would it be possible to post the model definition, data shapes as well as the used GPU, so that we can check the called kernels?

**Comment by konioyxgq** [comment]
I also tried it in cudnn8.0.2 and had the same problem. Sorry I cannot provide our model definition. If this is a definite problem, then I think the problem should be reproduced well.
In addition, I plan to train several times to find the best result, then find the corresponding convolution operator, modify the pytorch source code, and fix the convolution operator. Is this method reliable?
